core/blockchain.go

```diff
@@ -3519,40 +3543,207 @@ func (bc *BlockChain) PreviousCanonicalCoincidentOnPath(header *types.Header, sl
 	}
 }
 
-// RollbackToCanonical ensures that we are building on a canonical peg.
-func (bc *BlockChain) RollbackToCanonical(header *types.Header) error {
-	lastUncleHeader := &types.Header{}
-	lastUncleHash := common.Hash{}
-	for {
-		status := bc.domClient.GetBlockStatus(context.Background(), header)
-		// If the header is cononical break else keep looking
-		log.Debug("Status for CheckCanonical", "status", status, "number", header.Number)
-		switch status {
-		case quaiclient.CanonStatTy:
-			if (lastUncleHash != common.Hash{}) {
-				bc.ReOrgRollBack(lastUncleHeader, []*types.Header{}, []*types.Header{})
-			}
-			return nil
-		default:
-			lastUncleHeader = header
-			lastUncleHash = header.Hash()
+// The purpose of the Previous Coincident Reference Check (PCRC) is to establish
+// that we have linked untwisted chains prior to checking HLCR & applying external state transfers.
+// NOTE: note that it only guarantees linked & untwisted back to the prime terminus, assuming the
+// prime termini match. To check deeper than that, you need to iteratively apply PCRC to get that guarantee.
+func (bc *BlockChain) PCCRC(header *types.Header, headerOrder int) (types.PCRCTermini, error) {
```

Comment:
I'm confused as to why this duplication is necessary.

Reply:
Why is PCCOP needed instead of PCOP here? I understand that PCCOP returns an error on finding canonical but it feels like extra steps.